modules/geo-layers/src/mvt-layer/mvt-layer.js

```diff
@@ -216,12 +216,29 @@ export default class MVTLayer extends TileLayer {
     return renderedFeatures;
   }
 
+  getVisibleTiles() {
+    return this.state.tileset.selectedTiles;
+  }
+
+  _convertTilesToWGS84(tiles, viewport) {
+    tiles.forEach(tile => {
+      tile.transformToWorld = data => {
+        if (Array.isArray(data)) {
+          return data.map(object => transformTileCoordsToWGS84(object, tile, viewport));
+        }
+        return [];
+      };
+    });
+  }
+
   _onViewportChange() {
     const {onViewportChange} = this.props;
     if (onViewportChange) {
       const {viewport} = this.context;
+      this._convertTilesToWGS84(this.state.tileset.selectedTiles, viewport);
```

Comment:
You are still adding the transform conditionally (only when the user specifies `onViewportChange`. Move it to after `super.updateState`